lib/tester.coffee

```diff
@@ -98,20 +111,46 @@ class Tester
         dataCount = 0
         groups = []
         groupCount = 0
+        brackets = 0
+        listeningForIps = false
         @outs[portName].removeAllListeners()
+
+        finish = =>
+          @outs[portName].removeAllListeners()
+          data = data[0] if dataCount is 1
+          done data, groups, dataCount, groupCount if done
+          resolve data
+
         @outs[portName].on 'data', (packet) ->
+          return if listeningForIps
           data.push packet
           dataCount++
         @outs[portName].on 'begingroup', (group) ->
+          return if listeningForIps
           # Capture only unique groups
           unless groups.indexOf(group) isnt -1
             groups.push group
             groupCount++
         @outs[portName].on 'disconnect', =>
-          @outs[portName].removeAllListeners()
-          data = data[0] if dataCount is 1
-          done data, groups, dataCount, groupCount if done
-          resolve data
+          return if listeningForIps
+          finish()
+
+        @outs[portName].on 'ip', (packet) ->
+          listeningForIps = true
```

Comment:
Flagging like this isn't very reliable because the port always emits multiple events at a time, like `data` and `ip` and there may be a race between them. Is there some other way to detect which interface to use?